src/main/osd/msp_server_osd.c

```diff
@@ -323,23 +312,36 @@ int mspServerCommandHandler(mspPacket_t *cmd, mspPacket_t *reply)
             break;
 
         case MSP_SET_VOLTAGE_METER_CONFIG: {
-            uint8_t i = sbufReadU8(src);
-            if (i >= MAX_VOLTAGE_METERS) {
+            int index = sbufReadU8(src);
+
+            if (index >= MAX_VOLTAGE_METERS) {
                 return -1;
             }
-            // FIXME use `i`, see MSP_VOLTAGE_METER_CONFIG
-            batteryConfig()->vbatscale = sbufReadU8(src);               // actual vbatscale as intended
-            batteryConfig()->vbatmincellvoltage = sbufReadU8(src);      // vbatlevel_warn1 in MWC2.3 GUI
-            batteryConfig()->vbatmaxcellvoltage = sbufReadU8(src);      // vbatlevel_warn2 in MWC2.3 GUI
-            batteryConfig()->vbatwarningcellvoltage = sbufReadU8(src);  // vbatlevel when buzzer starts to alert
+
+            voltageMeterConfig(index)->vbatscale = sbufReadU8(src);
+            voltageMeterConfig(index)->vbatresdivval = sbufReadU8(src);
+            voltageMeterConfig(index)->vbatresdivmultiplier = sbufReadU8(src);
             break;
         }
 
-        case MSP_SET_CURRENT_METER_CONFIG:
-            batteryConfig()->currentMeterScale = sbufReadU16(src);
-            batteryConfig()->currentMeterOffset = sbufReadU16(src);
-            batteryConfig()->currentMeterType = sbufReadU8(src);
+        case MSP_SET_CURRENT_METER_CONFIG: {
+            int index = sbufReadU8(src);
+
+            if (index >= MAX_CURRENT_METERS) {
+                return -1;
+            }
+
+            currentMeterConfig(index)->currentMeterScale = sbufReadU16(src);
+            currentMeterConfig(index)->currentMeterOffset = sbufReadU16(src);
+            break;
+        }
+
+        case MSP_SET_BATTERY_CONFIG:
+            batteryConfig()->vbatmincellvoltage = sbufReadU8(src);      // vbatlevel_warn1 in MWC2.3 GUI
+            batteryConfig()->vbatmaxcellvoltage = sbufReadU8(src);      // vbatlevel_warn2 in MWC2.3 GUI
+            batteryConfig()->vbatwarningcellvoltage = sbufReadU8(src);  // vbatlevel when buzzer starts to alert
             batteryConfig()->batteryCapacity = sbufReadU16(src);
+            batteryConfig()->currentMeterSource = sbufReadU8(src);
```

Comment:
Yes, I am probably over-complicating this ... It may make sense to have separate batteries for controller/motors/failsafe in long term, but that is definitely not a priority.